tiledb/sm/query/reader.h

```diff
@@ -424,6 +424,18 @@ class Reader {
    */
   Status set_layout(Layout layout);
 
+  /**
+   * This is applicable only to dense arrays (errors out for sparse arrays),
+   * and only in the case where the array is opened in a way that all its
+   * fragments are sparse. If the input is `true`, then the dense array
+   * will be read in "sparse mode", i.e., the sparse read algorithm will
+   * be executing, returning results only for the non-empty cells.
```

Comment:
Can you briefly document here when sparse mode is used? It's clear what the meaning is, but not why it is needed.